modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -843,6 +963,284 @@ static void run_arithm_s(DST out[], const SRC in[], int width, int chan,
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - 3 * nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+template<>
+CV_ALWAYS_INLINE int absdiffc_simd_c3_impl<uchar>(const uchar in[], uchar out[],
+                                                  const v_float32& s1, const v_float32& s2,
+                                                  const v_float32& s3, const int length)
+{
+    constexpr int nlanes = static_cast<int>(v_uint8::nlanes);
+
+    if (length < 3 * nlanes)
+        return 0;
+
+    int x = 0;
+    v_float32 vectors[12];
+    for (;;)
+    {
+        for (; x <= length - 3 * nlanes; x += 3 * nlanes)
+        {
+            for (int i = 0; i < 12; ++i)
+            {
+                vectors[i] = v_cvt_f32(vx_load_expand_q(reinterpret_cast<const uchar*>(in + x + i * nlanes / 4)));
+            }
+
+            vx_store(reinterpret_cast<uchar*>(&out[x]), v_pack_u(v_pack(v_round(v_absdiff(vectors[0], s1)),
+                                                                        v_round(v_absdiff(vectors[1], s2))),
+                                                                 v_pack(v_round(v_absdiff(vectors[2], s3)),
+                                                                        v_round(v_absdiff(vectors[3], s1)))));
+
+            vx_store(reinterpret_cast<uchar*>(&out[x + nlanes]), v_pack_u(v_pack(v_round(v_absdiff(vectors[4], s2)),
+                                                                                 v_round(v_absdiff(vectors[5], s3))),
+                                                                          v_pack(v_round(v_absdiff(vectors[6], s1)),
+                                                                                 v_round(v_absdiff(vectors[7], s2)))));
+
+            vx_store(reinterpret_cast<uchar*>(&out[x + 2 * nlanes]), v_pack_u(v_pack(v_round(v_absdiff(vectors[8], s3)),
+                                                                                     v_round(v_absdiff(vectors[9], s1))),
+                                                                              v_pack(v_round(v_absdiff(vectors[10], s2)),
+                                                                                     v_round(v_absdiff(vectors[11], s3)))));
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - 3 * nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_c3(const T in[], const float scalar[], T out[], int width)
+{
+    constexpr int chan = 3;
+    int length = width * chan;
+
+    float init[6] = { *scalar, *(scalar + 1), *(scalar + 2), *scalar,
+                      *(scalar + 1), *(scalar + 2) };
+
+    v_float32 s1 = v_set_scalar<3>(scalar);
```

Comment:
I think it would be better to extend `init` array to `v_float::nlanes +2` and than just load
`s1 =vx_load(init+0)`
For 2 and 4 channels you could use the same approach or try `vx_lut_pairs/vx_lut_quads(scalar, vx_setzero_s32())` whatever show better performance

Reply:
agreed about simplifying/minimization of initialization code (no real performance impact)

Reply:
> I think it would be better to extend `init` array to `v_float::nlanes +2` and than just load
> `s1 =vx_load(init+0)`

Thanks for advice. I'll try.

> For 2 and 4 channels you could use the same approach or try `vx_lut_pairs/vx_lut_quads(scalar, vx_setzero_s32())` whatever show better performance

try `vx_lut_pairs/vx_lut_quads(scalar, vx_setzero_s32())` 

It is not so good idea. vx_lut_pairs() calls (pefix)_i32gather_epi64() intrinsic that have latency about 25. For comparison, vx_load() has a latency something about 8.

Reply:
And `vx_lut_quads()` has summary latency about equals to 33 vs latency 7 for xv_load().